### ISSUE TITLE
elpy and AucTeX - Possible compatibility problem.

### ISSUE THREAD

**Issue by richardfieldsend** [jorgenschaefer/elpy#1262]
Hi,
I hit this issue a while back, but revisited it recently, and hit the same issue.  I'm having some difficulty explaining exactly where the issue comes from (on my work machine it works fine, on my home machine it gives the error described below).

So the issue is that if I install elpy (along with the other programs, jedi etc) then the Python editing works exactly as expected. 

However, on my home machine, if I try and write a LaTeX document using AucTeX when elpy is activated then I get an error:

load: Symbol’s value as variable is void: LaTeX-dialect

The thing that is really strange is that I have both machines set up using Linux Mint (the same version 18.3), the same window manager (i3), and I keep my emacs initialisation file as a git repository which I copy between the machines using Git.

I have built Emacs from source, and almost all of the plugins (Magit, AucTeX, ebib etc) that I have installed I build from the Git repositories.
Any thoughts on what I can look at to fix this.  If I persist I can continue editing the document, but the message does come up regularly.

**Comment by galaunay** [comment]
Hi,
This is indeed strange, I also use elpy and auctex, but have never seen this.


Could you enable debugging (with `toggle-debug-on-error`) and post the full Backtrace here ?

**Comment by richardfieldsend** [comment]
Hi.
The backtrace when I create a LaTeX file and start editing it is shown below:

Debugger entered--Lisp error: (void-variable LaTeX-dialect)
  (TeX-add-style-hook "article" #f(compiled-function () #<bytecode 0xb6ae75>) LaTeX-dialect)
  load("/home/rf343/.emacs.d/elpa/auctex-11.89.7/style/article.elc" nil nil t)
  load-file("/home/rf343/.emacs.d/elpa/auctex-11.89.7/style/article.elc")
  TeX-load-style-file("/home/rf343/.emacs.d/elpa/auctex-11.89.7/style/article")
  TeX-load-style("article")
  #f(compiled-function (style) #<bytecode 0xd0c82d>)("article")
  mapcar(#f(compiled-function (style) #<bytecode 0xd0c82d>) ("latex2e" "art10" "article" "a4paper"))
  TeX-run-style-hooks("latex2e" "art10" "article" "a4paper")
  apply(TeX-run-style-hooks ("latex2e" "art10" "article" "a4paper"))
  TeX-auto-apply-entry((styles TeX-auto-file TeX-run-style-hooks))
  mapcar(TeX-auto-apply-entry (("index-entry" LaTeX-auto-index-entry LaTeX-add-index-entries LaTeX-index-entry-list LaTeX-index-entry-changed) ("bibliography" LaTeX-auto-bibliography LaTeX-add-bibliographies LaTeX-bibliography-list LaTeX-bibliography-changed) ("environment" LaTeX-auto-environment LaTeX-add-environments LaTeX-environment-list LaTeX-environment-changed) ("bibitem" LaTeX-auto-bibitem LaTeX-add-bibitems LaTeX-bibitem-list LaTeX-bibitem-changed) ("label" LaTeX-auto-label LaTeX-add-labels LaTeX-label-list LaTeX-label-changed) ("symbol" TeX-auto-symbol TeX-add-symbols TeX-symbol-list TeX-symbol-changed) (styles TeX-auto-file TeX-run-style-hooks)))
  TeX-auto-apply()
  TeX-update-style()
  TeX-arg-document(nil)
  TeX-parse-argument(nil TeX-arg-document)
  TeX-parse-arguments((TeX-arg-document))
  TeX-parse-macro("documentclass" (TeX-arg-document))
  TeX-insert-macro("documentclass")
  LaTeX-env-document("document")
  apply(LaTeX-env-document "document" nil)
  LaTeX-environment-menu("document")
  LaTeX-environment(nil)
  funcall-interactively(LaTeX-environment nil)
  call-interactively(LaTeX-environment nil nil)
  command-execute(LaTeX-environment)

This occurs when I start the file, then use C-e, select document, article and hit return (these are the defaults for  a new document). The error is then displayed.

**Comment by galaunay** [comment]
I have to say that I struggle to understand why Elpy would prevent latex for loading correctly one of its constants...

I wasn't able to reproduce the bug by using your emacs config (I assumed it was this [one](https://github.com/richardfieldsend/emacsLisp/tree/dc1211575f496d7529b02859887584c1add4747a)).
I can just tell you what I would do in your situation:
- Check what happen if you load elpy after latex.
- Check the message buffer at startup for latex error/warning messages (if you didn't already).
- Try to reload the latex file containing `LaTeX-dialext` at the end of your configuration with `(require 'latex)`, as a potential workaround.
- Try to redefine `LaTeX-dialect` at the end of your configuration with `(defconst LaTeX-dialect :latex)`, as a potential workaround.

Sorry for not being able to help you more.

**Comment by richardfieldsend** [comment]
Follow-up (and a huge thank you to galaunay!).

I got a chance to look at this on Sunday, and spent a while looking at what Emacs extensions were installed and how.  I found that there was a newer version of Elpy, so I installed that and removed the old one, then tried again...

I had also been looking at whether I could build Elpy from source, but didn't do so in the end.

Anyway, after much fiddling around I tried the Elpy support, and it was working as expected.  I then tried AucTeX and I am no longer receiving the error.  

I can't say what change I made fixed it, but I can now write LaTeX documents *and* code in Python.

Once again, thanks to galuanay for taking the time to look at the issue.
